regulations/parsing.py

```diff
@@ -0,0 +1,192 @@
+            "Content Files",
+        ],
+        "Rule": [
+            "Document ID",
+            "Document Type",
+            "Posted Date",
+            "Title",
+            "Topics",
+            "Content Files",
+        ],
+        "Proposed Rule": [
+            "Document ID",
+            "Document Type",
+            "Posted Date",
+            "Title",
+            "Content Files",
+        ],
+        # "Public Submission": ["Comment on Document ID", "Document Type", "Posted Date", "Comment", "Attachment Files"],
+        "Supporting & Related Material": [
+            "Document ID",
+            "Document Type",
+            "Posted Date",
+            "Content Files",
+        ],
+    }
+
+    rename_fields = {
+        "Notice": {},
+        "Rule": {},
+        "Proposed Rule": {},
+        # "Public Submission": {"Comment on Document ID": "Document ID"},
+        "Supporting & Related Material": {},
+    }
+
+    doc_type = record["Document Type"]
+    if doc_type in keep_fields and doc_type in rename_fields:
+        parsed_record = parse(
+            record,
+            keep_fields=keep_fields[doc_type],
+            rename_fields=rename_fields[doc_type],
+        )
+        parsed_record = {
+            k: (postprocess_fns[k](v) if k in postprocess_fns else v)
+            for k, v in parsed_record.items()
+        }
+
+        doc_id = parsed_record["Document ID"]
+        return doc_id, parsed_record
+    else:
+        return None
+
+
+def parse(
+    record: Dict[str, str],
+    keep_fields: List[str] = [],
+    rename_fields: Dict[str, str] = {},
+):
+    filtered_record = {k: v for k, v in record.items() if k in keep_fields}
+    renamed_filtered_record = {
+        (rename_fields[k] if k in rename_fields else k): v
+        for k, v in filtered_record.items()
+    }
+    return renamed_filtered_record
+
+
+def parse_files(files: Optional[str]):
+    return (
+        [{"URL": f, "File Type": os.path.splitext(f)[1]} for f in files.split(",")]
+        if files is not None
+        else []
+    )
+
+
+def parse_topics(topics: Optional[str]):
+    return topics.split(",") if topics is not None else []
+
+
+def parse_notice(record: Dict[str, str]):
+    parsed_record = {
+        k: v
+        for k, v in record.items()
+        if k in ["Document ID", "Posted Date", "Title", "Content Files"]
+    }
+    parsed_record["Content Files"] = parse_files(parsed_records["Content Files"])
+    return parsed_record
+
+
+def parse_rule(record: Dict[str, str]):
+    parsed_record = {
+        k: v
+        for k, v in record.items()
+        if k in ["Document ID", "Posted Date", "Title", "Content Files"]
+    }
+    return Rule(
+        document_id=record["Document ID"],
+        posted_date=record["Posted Date"],
+        title=record["Title"],
+        topics=parse_topics(record["Topics"]),
+        content_files=parse_files(record["Content Files"]),
+    )
+
+
+def parse_proposed_rule(record: pd.Series):
```

Comment:
Where are these functions used? It it like auto dispatched? I don't see references to them anywhere.